osv2/bulk.go

```diff
@@ -141,20 +144,50 @@ func (r *BulkRequest) Do(ctx context.Context, client *opensearch.Client) (*opens
 		return nil, err
 	}
 
-	resp := BulkResponse{}
-
-	if err := json.Unmarshal(respBuf.Bytes(), &resp); err != nil {
+	resp, err := r.parseResponse(respBuf)
+	if err != nil {
 		return nil, err
 	}
 
 	return &opensearchtools.OpenSearchResponse[BulkResponse]{
 		StatusCode:        osResp.StatusCode,
 		Header:            osResp.Header,
-		Response:          resp,
+		Response:          *resp,
 		ValidationResults: vrs,
 	}, nil
 }
 
+// parseResponse parses the response based on the bulk response settings
+func (r *BulkRequest) parseResponse(respBuf bytes.Buffer) (*BulkResponse, error) {
+	// Parse the complete response even if there are errors or not.
+	resp := BulkResponse{}
+	if !r.OnlyErrorResponse {
+		if err := json.Unmarshal(respBuf.Bytes(), &resp); err != nil {
+			return nil, err
+		}
+		return &resp, nil
+	}
+
+	// Parse first the errors and if errors is true parse the complete response.
+	respErr := BulkResponseOnlyErrors{}
```

Comment:
Like I mentioned below, this doesn't need to be an explicit named type.  You can declare and use it in one shot here.
```
var tmpResponse struct {
    Took.  bool `json:"took"`
    Errors bool `json:"errors"`
}
if err := json.Unmarshal(data, &tmpResponse); err != nil {
   return nil, err
}
...
```